lib/urls.js

```diff
@@ -13,7 +13,12 @@ module.exports = {
     view: '/e/:establishmentId/people/:profileId',
     invite: '/e/:establishmentId/people/invite',
     invitations: '/e/:establishmentId/people/invitations',
-    permission: '/e/:establishmentId/people/:profileId/permission'
+    permission: '/e/:establishmentId/people/:profileId/permission',
+    role: {
+      apply: '/e/:establishmentId/people/:profileId/role',
+      confirm: '/e/:establishmentId/people/:profileId/role/confirm',
+      success: '/e/:establishmentId/people/:profileId/role/success'
+    }
```

Comment:
🤔

Reply:
I'm not sure we should be defining urls externally to this level of granularity since there's never any need for other pages to link directly to anything other than the entry point.

Now I see it's been done everywhere else though...

Reply:
It makes the route builder nice to use :)

Reply:
I'm not sure I buy that - there's not _a lot_ of reason to use the route builder at all when you're not linking to pages.

Reply:
It is what it is though, so I might just need to deal with not liking it.

Reply:
e.g. https://github.com/UKHomeOffice/asl-pages/blob/master/pages/profile/roles/index.js#L36

Reply:
The problem is that it means that the url suffixes [like this](https://github.com/UKHomeOffice/asl-pages/blob/master/pages/profile/roles/index.js#L39) are coupled to the definition in the implementation url config, which means that neither can change independently. If the url builder allowed for a base url, and a sub-path defined at call time, then all of the url configuration would be encapsulated within the page, and wouldn't leak out into implementation - and so it would be able to change independently of implementation if we ever wanted to change journeys.